fastparquet/api.py

```diff
@@ -261,8 +261,18 @@ def statistics(self):
     @property
     def key_value_metadata(self):
         if self._kvm is None:
-            self._kvm = {k.key.decode(): k.value.decode()
-                for k in self.fmd.key_value_metadata or []}
+            kvm = {}
+            for kv in self.fmd.key_value_metadata or []:
+                try:
+                    key = kv.key.decode()
+                except UnicodeDecodeError:
+                    key = kv.key
```

Comment:
I wonder whether it should simply be rejected (same below), omitted or error. Can you try if other parquet frameworks (arrow, spark, older fastparquet) can recover non-utf8 bytestrings?